### PR TITLE
Offline logs feature

### PR DESCRIPTION
**Pull request by mm318**
*Description of changes:*

The changes implement the Cloudwatch offline feature.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cloudwatch_logs_common/CMakeLists.txt [hunk 1]
```diff
@@ -1,32 +1,58 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(cloudwatch_logs_common)
-set(CLOUDWATCH_LOGS_COMMON_VERSION 1.0.0)
+set(CLOUDWATCH_LOGS_COMMON_VERSION 1.0.0) #todo
```
Comment (dabonnie) [inline cloudwatch_logs_common/CMakeLists.txt]: Version should increase (and remove todo)

File: cloudwatch_logs_common/CMakeLists.txt [hunk 2]
```diff
@@ -60,21 +87,37 @@ INSTALL(EXPORT ${PROJECT_NAME}-targets DESTINATION share/${PROJECT_NAME}/cmake)
...
-  ${PROJECT_SOURCE_DIR}/test/include
-)
-target_link_libraries(test_cloudwatch_logs_common
-  ${PROJECT_NAME}
-  ${aws_common_LIBRARIES}
-  ${GMOCK_LIBRARY}
-  pthread
-)
+if(GMOCK_LIBRARY)
+  message("Building tests")
```
Comment (dabonnie) [inline cloudwatch_logs_common/CMakeLists.txt]: Do we need this message?

File: cloudwatch_logs_common/CMakeLists.txt (hunk 1 above)
Comment (mm318) [reply cloudwatch_logs_common/CMakeLists.txt]: Bumped the minor version and removed the comment.

File: cloudwatch_logs_common/CMakeLists.txt (hunk 2 above)
Comment (mm318) [reply cloudwatch_logs_common/CMakeLists.txt]: Removed the message printout.

**Pull request merged**